src/js/ripple/amount.js

```diff
@@ -453,56 +322,37 @@ Amount.prototype.invert = function() {
  * @returns {Amount}
  * @throws {Error} if offset exceeds legal ranges, meaning the amount value is bigger than supported
  */
-Amount.prototype.canonicalize = function() {
-  if (!(this._value instanceof BigInteger)) {
-    // NaN.
-    // nothing
-  } else if (this._is_native) {
-    // Native.
-    if (this._value.equals(BigInteger.ZERO)) {
-      this._offset      = 0;
-      this._is_negative = false;
-    } else {
-      // Normalize _offset to 0.
-
-      while (this._offset < 0) {
-        this._value  = this._value.divide(Amount.bi_10);
-        this._offset += 1;
-      }
-
-      while (this._offset > 0) {
-        this._value  = this._value.multiply(Amount.bi_10);
-        this._offset -= 1;
-      }
-    }
 
-  } else if (this.is_zero()) {
-    this._offset      = Amount.cMinOffset;
-    this._is_negative = false;
+Amount.prototype.canonicalize = function(roundingMode) {
+  if (this._is_native) {
+    var value = this._value.round(6, BigNumber.ROUND_DOWN);
+    this._value = value.absoluteValue().greaterThan(Amount.bi_xrp_max)
```

Comment:
I think what sublimator is saying is that this change introduces _new_ behavior which is both incorrect and very dangerous - numerical overflow is handled by setting the value to 0 instead of either setting it to NaN or throwing an exception.

Correctness is of supreme importance - this is financial software, and people will be putting serious money through this - so it's _always_ a relevant issue worth bringing up if you see code that you even suspect is incorrect.